### PR TITLE
Strict build validation for rule identifiers/CCEs

### PR DESCRIPTION
**Pull request by cipherboy**
#### Description:

- Implements stricter validation of identifiers, refusing to build if validations fail:
  - Refuses to build if CCE is empty
  - Refuses to build if CCE is invalid
- Fixes existing rules to conform to these new standards.

### CODE REVIEW

File: shared/utils/yaml-to-shorthand.py [hunk 1]
```diff
@@ -494,7 +494,7 @@ def from_yaml(yaml_file, product_yaml=None):
         rule.severity = required_yaml_key(yaml_contents, "severity")
         del yaml_contents["severity"]
         rule.references = yaml_contents.pop("references", [])
-        rule.identifiers = yaml_contents.pop("identifiers", [])
```
Comment (mpreisler) [inline shared/utils/yaml-to-shorthand.py]: it's not clear to me why this change was needed, it seems like defaulting to [] and having the assurance that rule.identifiers is always a list() would be OK and would work with the rest of the code

File: shared/utils/yaml-to-shorthand.py [hunk 2]
```diff
@@ -508,8 +508,27 @@ def from_yaml(yaml_file, product_yaml=None):
             raise RuntimeError("Unparsed YAML data in '%s'.\n\n%s"
                                % (yaml_file, yaml_contents))
 
+        if not rule.validate_identifiers(yaml_file):
```
Comment (mpreisler) [inline shared/utils/yaml-to-shorthand.py]: this will never return False :-) Just running it is fine.

File: shared/utils/fix-cces.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    elif sys.argv[1] == 'invalid':
+        find_invalid_cce(sys.argv[2])
+    else:
+        print("Usage: %s mode /full/path/to/src/directory" % sys.argv[0])
+        print("Modes:")
+        print("\tempty - check and fix rules with empty identifiers")
+        print("\tinvalid - find rules with invalid identifiers")
+
+if __name__ == "__main__":
+    __main__()
```
Comment (mpreisler) [inline shared/utils/fix-cces.py]: hmm, this is a cool script and might help us write other scripts but on the other hand it is a one-off, I don't expect anyone to ever run it again because build system now forbids invalid identifiers so there is no way to introduce them

File: shared/utils/yaml-to-shorthand.py (hunk 1 above)
Comment (mpreisler) [reply shared/utils/yaml-to-shorthand.py]: discussed this with @cipherboy , turns out it needs to be a dict()

**Comment by cipherboy** [comment]
Shorter PEP8 violation list now. :)

File: shared/utils/yaml-to-shorthand.py [hunk 4]
```diff
@@ -508,8 +508,25 @@ def from_yaml(yaml_file, product_yaml=None):
             raise RuntimeError("Unparsed YAML data in '%s'.\n\n%s"
                                % (yaml_file, yaml_contents))
 
+        rule.validate_identifiers(yaml_file)
         return rule
 
+    def validate_identifiers(self, yaml_file):
+        if self.identifiers is None:
```
Comment (mpreisler) [inline shared/utils/yaml-to-shorthand.py]: this will never happen

Comment (cipherboy) [reply shared/utils/yaml-to-shorthand.py]: @mpreisler Updated

**Comment by mpreisler** [comment]
Crap, I merged the other PR and it created a conflict. This needs a rebase now.

**Comment by cipherboy** [comment]
@mpreisler Rebased.

**Comment by mpreisler** [comment]
ACK
